hedera-node/src/main/java/com/hedera/services/utils/MapValueListUtils.java

```diff
@@ -104,21 +109,57 @@ <K, V extends FastCopyable> K overwritingRemoveFromMapValueList(
 			@NotNull final K root,
 			@NotNull final MapValueListMutation<K, V> listRemoval
 	) {
-		return internalRemoveFromMapValueList(key, root, listRemoval, false);
+		return internalMutateFromMapValueList(key, root, listRemoval, false, true, false);
+	}
+
+	/**
+	 * Unlinks the value of the given key from its containing linked list in the map represented by the
+	 * given {@link MapValueListMutation}, updating the doubly-linked list to maintain the prev/next keys of the
+	 * "adjacent" value(s) as needed and resets the next and prev pointers of this value.
+	 * Does <i>not</i> use {@link MapValueListMutation#getForModify(Object)}.
```

Comment:
```suggestion
	 * Uses {@link MapValueListMutation#getForModify(Object)}.
```

Reply:
We are not using it Neeha, if we are .. then we should revert it.

Reply:
I mean the third parameter is `true` for this method `useGetForModify`